test/tools/spec-runner/index.js

```diff
@@ -666,6 +685,74 @@ const kOperations = new Map([
       if (args.out) options.out = args.out;
       return collection.mapReduce(map, reduce, options);
     }
+  ],
+  [
+    'assertCollectionExists',
+    (operation, testRunnerContext, context /*, options */) => {
+      const collectionName = operation.arguments.collection;
+      const session = maybeSession(operation, context);
+      const db = context.client.db(operation.arguments.database);
+      return withoutCommandMonitoring(context.client, () =>
+        db
+          .listCollections({ session })
+          .toArray()
+          .then(results => results.map(({ name }) => name))
+          .then(collections => expect(collections).to.include(collectionName))
+      );
+    }
+  ],
+  [
+    'assertCollectionNotExists',
+    (operation, testRunnerContext, context /*, options */) => {
+      const collectionName = operation.arguments.collection;
+      const session = maybeSession(operation, context);
+      const db = context.client.db(operation.arguments.database);
+      return withoutCommandMonitoring(context.client, () =>
+        db
+          .listCollections({ session })
+          .toArray()
+          .then(results => results.map(({ name }) => name))
+          .then(collections => expect(collections).to.not.include(collectionName))
+      );
+    }
+  ],
+  [
+    'assertIndexExists',
+    (operation, testRunnerContext, context /*, options */) => {
+      const collectionName = operation.arguments.collection;
+      const indexName = operation.arguments.index;
+      const session = maybeSession(operation, context);
+      const db = context.client.db(operation.arguments.database);
+      return withoutCommandMonitoring(context.client, () =>
+        db
+          .collection(collectionName)
+          .listIndexes({ session })
+          .toArray()
+          .then(results => results.map(({ name }) => name))
+          .then(indexes => expect(indexes).to.include(indexName))
+      );
+    }
+  ],
+  [
+    'assertIndexNotExists',
+    (operation, testRunnerContext, context /*, options */) => {
+      const collectionName = operation.arguments.collection;
+      const indexName = operation.arguments.index;
+      const session = maybeSession(operation, context);
+      const db = context.client.db(operation.arguments.database);
+      return withoutCommandMonitoring(context.client, () =>
+        db
+          .collection(collectionName)
+          .listIndexes({ session })
+          .toArray()
+          .then(results => results.map(({ name }) => name))
+          .then(indexes => expect(indexes).to.not.include(indexName))
+      ).catch(err => {
+        if (err.code !== MONGODB_ERROR_CODES.NamespaceNotFound) {
+          throw err;
+        }
+      });
+    }
```

Comment:
I saw that these are part of the unified test runner – is this okay to add here like this? Maybe porting the CSFLE tests over isn’t that hard, but it felt like that would be out of scope here.